src/ws/session.c

```diff
@@ -544,6 +544,17 @@ perform_gssapi (void)
   setenv ("KRB5CCNAME", "FILE:/dev/null", 1);
   setenv ("KRB5RCACHETYPE", "none", 1);
 
+  /* Always attempt to use gss-proxy as it is transparent with the interposer
+   * plugin in MIT Kerberos. If interposer plugin is not available, it is no-op,
+   * if gss-proxy is not running, it is no-op either.
+   *
+   * No error messages will be in the logs for case when gss-proxy is not used.
+   *
+   * Currently gss-proxy interposer plugin uses secure_getenv() to read the environment
+   * variables and it will not work with cockpit-session due to setuid bit, cockpit-session
+   * needs to implement fork-and-drop-privileges way of handling GSSAPI. */
```

Comment:
I discussed this again with @simo5 and he is opposed strongly to do getenv() in the interposer plugin. This means we either need to be able to pass KRB5_KTNAME variable from the cockpit-ws environment or have some way to set it for cockpit-session specifically. For example, /etc/cockpit/cockpit.conf could be used for that.

At the very least I want to get KRB5_KTNAME=/etc/httpd/alias/ipa.keytab set when cockpit is running on IPA master to avoid duplication of the HTTP/fqdn@REALM keys in /etc/krb5.keytab.